lib/datadog/statsd/serialization/stat_serializer.rb

```diff
@@ -43,18 +33,21 @@ def global_tags
         end
 
         private
+
         attr_reader :prefix
         attr_reader :tag_serializer
 
         def formated_name(name)
-          formated = name.is_a?(String) ? name.dup : name.to_s
-
-          formated.tap do |f|
-            # replace Ruby module scoping with '.'
-            f.gsub!('::', '.')
-            # replace reserved chars (: | @) with underscores.
-            f.tr!(':|@', '_')
+          if name.is_a?(String)
+            # DEV: gsub is faster than dup.gsub!
```

Comment:
👍

Reply:
This is interesting given the extra allocation(?) in `gsub!`, do we know why?